### PR TITLE
google-music-scripts: init at 3.0.0

### PR DESCRIPTION
**Pull request by jbaum98**
###### Motivation for this change

Add `google-music-scripts`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/bidict/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/bidict/default.nix]:
```suggestion
    license = licenses.mpl20;
```

This will fix eval. Licenses attrs can be seen https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix

File: pkgs/development/python-modules/pprintpp/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/pprintpp/default.nix]:
```suggestion
    ${python.interpreter} test.py
```

File: pkgs/development/python-modules/audio-metadata/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/audio-metadata/default.nix]: I don't see python2 support declared https://github.com/thebigmunch/audio-metadata/blob/d53cf68258fba959663945625eefff4fcb413280/setup.py#L25

so it needs to be disabled for that interpreter with the `disabled` parameter.

Comment (worldofpeace) [reply pkgs/development/python-modules/audio-metadata/default.nix]: Actual constraint is here https://github.com/thebigmunch/audio-metadata/blob/d53cf68258fba959663945625eefff4fcb413280/pyproject.toml#L20

File: pkgs/development/python-modules/google-music-proto/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/google-music-proto/default.nix]: Needs `disabled` for constrained python interpreter versions https://github.com/thebigmunch/google-music-proto/blob/3f7f81cced963a01bc9dfbbc7cd44ca6092c3aa9/pyproject.toml#L20

File: pkgs/development/python-modules/click-default-group/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/click-default-group/default.nix]: Saw tests weren't distributed, this resolved that. Though comment is needed.

File: pkgs/tools/audio/google-music-scripts/default.nix
Comment (worldofpeace) [inline pkgs/tools/audio/google-music-scripts/default.nix]: I'd do `python3.pkgs.callPackage` for `google-music-scripts` as well since it's only for python 3.

Comment (worldofpeace) [reply pkgs/tools/audio/google-music-scripts/default.nix]: Also why not use `overridePythonAttrs` for `click6`?

File: pkgs/development/python-modules/bidict/default.nix
Comment (jbaum98) [reply pkgs/development/python-modules/bidict/default.nix]: Weird, it's been building fine for me locally.

File: pkgs/development/python-modules/audio-metadata/default.nix
Comment (jbaum98) [reply pkgs/development/python-modules/audio-metadata/default.nix]: Thank you! I couldn't figure out how to disable python2.

File: pkgs/development/python-modules/pprintpp/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/pprintpp/default.nix]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, python, nose, parameterized }:
```

for `${python.interpreter}`

File: pkgs/development/python-modules/bidict/default.nix
Comment (worldofpeace) [reply pkgs/development/python-modules/bidict/default.nix]: This is actually a special check that ofborg does. The results of the checks can be seen at the bottom of any pr.

File: pkgs/tools/audio/google-music-scripts/default.nix
Comment (worldofpeace) [inline pkgs/tools/audio/google-music-scripts/default.nix]: This is in the click package already for the exact purpose of overriding it to `6.7`. Any reason why it's here?

File: pkgs/tools/audio/google-music-scripts/default.nix
Comment (jtojnar) [inline pkgs/tools/audio/google-music-scripts/default.nix]: If older version of click is necessary, could you open an upstream bug asking for them to update to click 7?

File: pkgs/tools/audio/google-music-scripts/default.nix
Comment (jbaum98) [reply pkgs/tools/audio/google-music-scripts/default.nix]: You mean the version check? Sorry that was copy-pasted.

File: pkgs/tools/audio/google-music-scripts/default.nix
Comment (jbaum98) [reply pkgs/tools/audio/google-music-scripts/default.nix]: I realized that just by bumping the version number click 7 works fine, so I'll use a patch for that and submit it upstream.

Comment (jbaum98) [reply pkgs/tools/audio/google-music-scripts/default.nix]: thebigmunch/google-music-scripts#21

**Comment by jbaum98** [comment]
@worldofpeace Now that `google-music-scripts` has been updated to allow `click` version 7, I think this is ready to go.

**Comment by worldofpeace** [comment]
> @worldofpeace Now that `google-music-scripts` has been updated to allow `click` version 7, I think this is ready to go.

Nice, I'll give this a final review to see if there's anything amiss.

**Comment by worldofpeace** [comment]
Looks like https://github.com/NixOS/nixpkgs/pull/53782/commits/6bc8ca7f3040e31c32287a3aa96de3b06a8730be had a little fixup mishap

File: pkgs/development/python-modules/bidict/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/bidict/default.nix]: I don't see the test being ran in the build output.
Checked the tarball and they're at `tests`.

File: pkgs/development/python-modules/audio-metadata/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/audio-metadata/default.nix]: This package dosen't have any tests.

File: pkgs/development/python-modules/google-music-proto/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/google-music-proto/default.nix]: Not seeing any tests here also.

File: pkgs/development/python-modules/google-music/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/google-music/default.nix]: Not seeing tests here also.

**Comment by jbaum98** [comment]
So if a package has no tests we want to explicitly disable `checkPhase`?

**Comment by worldofpeace** [comment]
> So if a package has no tests we want to explicitly disable `checkPhase`?

Yep. Other python maintainers here also agree on this.

**Review by worldofpeace — APPROVED** [review]
Think it's good :+1: 
Thanks for your patience @jbaum98 

Haven't tested the actual function of the program though.

I'll merge this with the approval of one more maintainer and If someone can vouch for the function.

cc @dotlambda

**Comment by worldofpeace** [comment]
Note, if I don't get another approval within <24hrs I'll just merge it.

File: pkgs/development/python-modules/audio-metadata/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/audio-metadata/default.nix]:
```suggestion
  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/bidict/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/bidict/default.nix]: I'd prefer doing this in `postPatch` using `substituteInPlace`.
Did you submit a patch upstream?

File: pkgs/development/python-modules/google-music-proto/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/google-music-proto/default.nix]:
```suggestion
  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/google-music-utils/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/google-music-utils/default.nix]:
```suggestion
  disabled = !pythonOlder "3.6";
```

File: pkgs/development/python-modules/google-music-utils/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/google-music-utils/default.nix]: Please add a comment.

File: pkgs/development/python-modules/google-music/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/google-music/default.nix]:
```suggestion
  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/pprintpp/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pprintpp/default.nix]: You should submit that patch upstream and use `fetchpatch`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
I can confirm that this google-music-scripts is functioning correctly.

File: pkgs/development/python-modules/bidict/default.nix
Comment (jbaum98) [reply pkgs/development/python-modules/bidict/default.nix]: I opened a PR, but I actually realized that it's building without the patch.

File: pkgs/development/python-modules/pprintpp/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pprintpp/default.nix]: Don't use `let`. You might have to use parantheses.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Remove the patch files.

File: pkgs/development/python-modules/pprintpp/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/pprintpp/default.nix]: PR URLs are stable once merged. Use https://github.com/wolever/pprintpp/commit/873217674cc824b4c1cfdad4867c560c60e8d806.patch instead.

Comment (jbaum98) [reply pkgs/development/python-modules/pprintpp/default.nix]: What about https://github.com/wolever/pprintpp/pull/21.patch?

Comment (dotlambda) [reply pkgs/development/python-modules/pprintpp/default.nix]: That might change if you or upstream add commits or force-push.

Comment (jbaum98) [reply pkgs/development/python-modules/pprintpp/default.nix]: But won't it be what we want it to be? And if I force push won't that commit maybe go away?

Comment (jbaum98) [reply pkgs/development/python-modules/pprintpp/default.nix]: Oh but the `sha256` might change and that will break it.

**Comment by worldofpeace** [comment]
Maybe a little more than 24 hours :rofl:

**Comment by jbaum98** [comment]
No problem, thanks @worldofpeace and @dotlambda for your help!

**Pull request merged**
